core/src/cats/sql_create.cc

```diff
@@ -1256,9 +1256,12 @@ bool BareosDb::CreateNdmpEnvironmentString(JobControlRecord* jcr,
   EscapeString(jcr, esc_envvalue, value, strlen(value));
   Mmsg(cmd,
        "INSERT INTO NDMPJobEnvironment (JobId, FileIndex, EnvName, EnvValue)"
-       " VALUES ('%s', '%s', '%s', '%s')",
+       " VALUES ('%s', '%s', '%s', '%s')"
+       " ON CONFLICT ON CONSTRAINT ndmpjobenvironment_pkey"
```

Comment:
This relies on the constraint name, I guess using the column names is better.
The PostgreSQL docs say:
It is often preferable to use unique index inference rather than naming a constraint directly using ON CONFLICT ON CONSTRAINT constraint_name. Inference will continue to work correctly when the underlying index is replaced by another more or less equivalent index in an overlapping way, for example when using CREATE UNIQUE INDEX ... CONCURRENTLY before dropping the index being replaced.